lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb

```diff
@@ -2,468 +2,358 @@
+      end
+
+      if bc_product
+        payload[:id] = bc_product['id']
+      end
+
+
+      return @product_client.upsert(payload)
+    end
 
-        def update_meta_fields(upsert_fields, delete_fields)
-            meta_fields = []
+    # Manages the custom/meta fields for the product. Since we don't maintain a _link_ between the
+    # Custom/Meta Fields in BigCommerce and their associated data in Acumen, we don't have a very
+    # reliable way to fetch/update individual fields.
+    #
+    # Instead, this function fetches _all_ the product fields, and we use the field name to map
+    # them to the correct Acumen data.
+    #
+    # Additionally, BigCommerce does not allow _blank_ values in Custom/Meta Fields, so if field data
+    # is removed from Acumen, then we'll need to _delete_ the associated field in BigCommerce.
+    #
+    # It is also important to note that this function _will not_ set `related_product_ids`. That
+    # field is managed by a separate function.
+    def update_fields(raw_product, bc_product, mapper, map, client)
+      current_fields = client.get_for_product(bc_product['id'])
+      fields = mapper.map(map, raw_product, bc_product, current_fields, options['meta_fields_namespace'])
+
+      # Delete fields
+      fields[:delete].each do |field|
+          client.delete(field['product_id'], field['id'])
+      end
+
+      # Upsert fields
+      fields[:upsert].each do |field|
+          client.upsert(field['product_id'], field)
+      end
+
+      return fields
+    end
 
-            upsert_fields.each do |field|
-                meta_fields << @meta_field.upsert(field)
-            end
+    # This function manages the `related_product_ids` field which is used to link
+    # different product formats together.
+    # Each item in the provided array will have the following structure:
+    #
+    #     {
+    #       bc_product -- the BigCommerce product record
+    #       custom_fields -- Custom Fields in BigCommerce
+    #       meta_fields -- Meta Fields in BigCommerce
+    #       acumen_data -- The Acumen product record
+    #     }
+    #
+    def set_related_product_ids(product_data)
+      # TODO: The value of `related_product_ids` should be set to a CSV of all other formats.
+      # So, if products 1, 2, 3, and 4 are all different formats of the same title, Product 1 should
+      # have a value of `2,3,4`.
+      #
+      # Additionally, the custom_fields key in the hash should be _updated_ to include the
+      # related_product_ids field. And the updated hash should be returned.
+
+      product_ids = []
+
+      product_data.each do |product_hash|
+        product_ids.push(product_hash[:bc_product]['id'])
+      end
+
+      product_data.each do |product_hash|
+        product_ids.delete(product_hash[:bc_product]['id'])
+        product_hash[:custom_fields]['related_product_ids'] = product_ids * ','
+        product_ids.push(product_hash[:bc_product]['id'])
+      end
+
+      return product_data
```

Comment:
@abourge This isn't actually setting the custom field.

It's included on the output from the agent, but it's not getting published in BigCommerce